src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,334 @@
+
+        #region Paths
+
+        /// <summary>
+        /// Validate each path provided and if valid, add to array of paths to process.
+        /// If path is a literal path it is added to the array to process; we cannot validate them until we
+        /// try to process file contents.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="literalPath"></param>
+        /// <returns></returns>
+        private List<string> ResolvePaths(string[] path, bool literalPath)
+        {
+            List<string> pathsToProcess = new List<string>();
+            ProviderInfo provider = null;
+            PSDriveInfo drive = null;
+
+            foreach (string currentPath in path)
+            {
+                List<string> newPaths = new List<string>();
+
+                if (literalPath)
+                {
+                    newPaths.Add(Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(currentPath, out provider, out drive));
+                }
+                else
+                {
+                    try
+                    {
+                        newPaths.AddRange(Context.SessionState.Path.GetResolvedProviderPathFromPSPath(currentPath, out provider));
+                    }
+                    catch (ItemNotFoundException e)
+                    {
+                        if (!WildcardPattern.ContainsWildcardCharacters(currentPath))
+                        {
+                            ErrorRecord errorRecord = new ErrorRecord(e, "FileNotFound", ErrorCategory.ObjectNotFound, path);
+                            WriteError(errorRecord);
+                            continue;
+                        }
+                    }
+                }
+
+                if (!provider.Name.Equals("FileSystem", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Write a non-terminating error message indicating that path specified is not supported.
+                    string errorMessage = StringUtil.Format(UtilityCommonStrings.FormatHexOnlySupportsFileSystemPaths, currentPath);
+                    ErrorRecord errorRecord = new ErrorRecord(new ArgumentException(errorMessage),
+                                                                "FormatHexOnlySupportsFileSystemPaths",
+                                                                ErrorCategory.InvalidArgument,
+                                                                currentPath);
+                    WriteError(errorRecord);
+                    continue;
+                }
+
+                pathsToProcess.AddRange(newPaths);
+            }
+
+            return pathsToProcess;
+        }
+
+        /// <summary>
+        /// Pass each valid path on to process its contents.
+        /// </summary>
+        /// <param name="pathsToProcess"></param>
+        private void ProcessPath(List<string> pathsToProcess)
+        {
+            // If there are no paths to proccess, just return.
```

Comment:
Thank you.  Good point, it turns out we do not need this code.  It does throw an error if the path does not resolve to a file.  

```
PS > fhx -Path "c:\dlfkjasdkfj.txt", "c:\ldkfj.txt"
fhx : Cannot find path 'C:\dlfkjasdkfj.txt' because it does not exist.
At line:1 char:1
+ fhx -Path "c:\dlfkjasdkfj.txt", "c:\ldkfj.txt"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (System.String[]:String[]) [Format-Hex], ItemNotFoundException
    + FullyQualifiedErrorId : FileNotFound,Microsoft.PowerShell.Commands.FormatHex

fhx : Cannot find path 'C:\ldkfj.txt' because it does not exist.
At line:1 char:1
+ fhx -Path "c:\dlfkjasdkfj.txt", "c:\ldkfj.txt"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (System.String[]:String[]) [Format-Hex], ItemNotFoundException
    + FullyQualifiedErrorId : FileNotFound,Microsoft.PowerShell.Commands.FormatHex